NMCA_001/NMCA_001_script.lua

```diff
@@ -0,0 +1,767 @@
+    ----------
+    local M3Walls = ScenarioUtils.CreateArmyGroup('UEF', 'M3Walls')
+    local Engineers = ScenarioUtils.CreateArmyGroup('UEF', 'M3Engineers')
+
+    ----------
+    -- Run AI
+    ----------
+    M3BaseAI.UEFBaseAI()
+    ScenarioInfo.EnemyCommander = ScenarioUtils.CreateArmyUnit('UEF', 'Commander')
+    ScenarioInfo.EnemyCommander:SetCustomName("CDR Parker")
+
+    if Difficulty == 1 then
+        ScenarioInfo.EnemyCommander:CreateEnhancement('AdvancedEngineering')
+        ScenarioInfo.EnemyCommander:CreateEnhancement('LeftPod')
+        ScenarioInfo.EnemyCommander:CreateEnhancement('RightPod')
+    elseif Difficulty == 2 then
+        ScenarioInfo.EnemyCommander:CreateEnhancement('AdvancedEngineering')
+        ScenarioInfo.EnemyCommander:CreateEnhancement('Shield')
+        ScenarioInfo.EnemyCommander:CreateEnhancement('ResourceAllocation')   
+    elseif Difficulty == 3 then
+        ScenarioInfo.EnemyCommander:CreateEnhancement('AdvancedEngineering')
+        ScenarioInfo.EnemyCommander:CreateEnhancement('Shield')
+        ScenarioInfo.EnemyCommander:CreateEnhancement('HeavyAntiMatterCannon')
+    end
+
+    ----------
+    -- Spawn PBase
+    ----------
+    local PBase = ScenarioUtils.CreateArmyGroup('UEF', 'PowerBase')
+    local PBaseWalls = ScenarioUtils.CreateArmyGroup('UEF', 'M3PWalls')
+    ScenarioInfo.Shields = ScenarioUtils.CreateArmyGroup('UEF', 'M3Shields')
+
+    ----------
+    -- Spawn Patrols
+    ----------
+    local units = ScenarioUtils.CreateArmyGroupAsPlatoon('UEF', 'M3_AirPatrol_' .. Difficulty, 'NoFormation')
+
+    for k, v in units:GetPlatoonUnits() do
+        ScenarioFramework.GroupPatrolRoute({v}, ScenarioPlatoonAI.GetRandomPatrolRoute(ScenarioUtils.ChainToPositions('M3_AirPatrol_Chain')))
+    end
+
+    local AttackUnits = ScenarioUtils.CreateArmyGroupAsPlatoon('UEF', 'M3_Init_Attack_' .. Difficulty, 'AttackFormation')
+
+    for k, v in AttackUnits:GetPlatoonUnits() do
+        ScenarioFramework.GroupPatrolRoute({v}, ScenarioPlatoonAI.GetRandomPatrolRoute(ScenarioUtils.ChainToPositions('M3_LandAttack_Chain')))
+    end
+
+    local PBasePatrol = ScenarioUtils.CreateArmyGroupAsPlatoon('UEF', 'M3PBasePatrol', 'NoFormation')
+
+    for k, v in PBasePatrol:GetPlatoonUnits() do
+        ScenarioFramework.GroupPatrolRoute({v}, ScenarioPlatoonAI.GetRandomPatrolRoute(ScenarioUtils.ChainToPositions('M3_PBase_Patrol_Chain')))
+    end
+
+    ----------
+    -- Dialogue
+    ----------
+    ScenarioFramework.Dialogue(OpStrings.M2_Update_Commander, M2ObjectviesExtended, true)
+end
+
+function M2ObjectviesExtended()
+    ----------
+    -- Objectives
+    ----------
+    ScenarioInfo.M2P3 = Objectives.Capture(
+    'primary',                      -- type
+    'incomplete',                   -- complete
+    'Capture Enemy Tech Centre',  -- title
+    'We need to find out what is going on. Capture the enemy tech centre so we can access the data.',  -- description
+    {
+        MarkUnits = true,

```

Comment:
this line should not be needed, units for Capture objective are marked by default